### PR TITLE
Minor content changes

### PR DESCRIPTION
**Pull request by jordangov**
This PR makes numerous small content changes (projects, press, and join us mostly). The changes were requested by @MoVaheyDDS and are requested to be live by Thursday morning 10/25 for the DDS-day in Augusta.

@jgarber623-gov Please wait for Mo's 👍 (either by Slack or email) before merging/deploying.

### CODE REVIEW

**Comment by jordangov** [comment]
@jgarber623-gov Not sure why Travis/html-proofer/curl is being dumb, but the link works and html-proofer works on my machine. It's not a 30X issue (I get a 200 locally), so not sure what's going on. In any case, the preview URL on Netlify looks fine... if you can figure this one out great, if not... 🤷‍♂️

File: src/_data/press.yml [hunk 1]
```diff
@@ -23,9 +23,13 @@
    author: FedScoop
 
  - title: DoD announces “Hack the Pentagon” initiative
-   press_url: http://intelligencecommunitynews.com/dod-announces-hack-the-pentagon-initiative/
+   press_url: https://intelligencecommunitynews.com/dod-announces-hack-the-pentagon-initiative/
```
Comment (jgarber623-gov) [inline src/_data/press.yml]: 💪

File: src/_data/projects.yml [hunk 2]
```diff
@@ -33,9 +33,9 @@
...
\ No newline at end of file
+     The Defense Digital Service launched a program to engage and support technical talent within the U.S. Army Cyber Command. Through a series of collaborative projects, highly-trained army officers and soldiers work side-by-side with DDS civilians in unclassified, collaborative, startup-like spaces. Examples of projects include <a href="https://www.wired.com/story/pentagon-dream-team-tech-savvy-soldiers/">developing drone detection technologies</a>, hunting adversaries on DoD networks, and <a href="https://www.fedscoop.com/defense-digital-service-redesigns-army-cyber-command-training/">redesigning training for cyber soldiers</a>.<br><strong>Active Duty Military are encouraged to apply for short-term technical positions under DDS. Military can apply to the 'JYN' program on <a href="https://hire.withgoogle.com/public/jobs/ddsmil/view/P_AAAAAAEAAF3KkxWUnRXoHq">Google Hire</a>.</strong>
```
Comment (jgarber623-gov) [inline src/_data/projects.yml]: Couple of comments here:

1. 

> highly-trained army officers and soldiers

Shouldn't that be "Army" uppercase?

2. We inconsistently use "DOD" and "DoD" in the copy across the site. Which is most accurate?

3. `'JYN'` should probably be `"JYN"`.

File: src/_data/press.yml (hunk 1 above)
Comment (jordangov) [reply src/_data/press.yml]: 🍾

File: src/_data/projects.yml (hunk 2 above)
Comment (jordangov) [reply src/_data/projects.yml]: These are questions for Mo... I'd ask in Slack (she's not really a GH user)

**Pull request merged**